File: src/main/java/com/ggardet/hospital/internal/entity/Treatment.java
package com.ggardet.hospital.internal.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Data;

import java.util.UUID;

@Entity
@Data
@Table(name = "treatment", schema = "hospitals")
public class Treatment {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(nullable = false)
    private String name;

    @Column
    private String description;

    @Column
    private int durationMinutes;

    @Column
    private String protocol;

    @ManyToOne
    @JoinColumn(name = "hospital_id")
    private Hospital hospital;
}

File: src/main/kotlin/com/ggardet/hospital/internal/entity/Treatment.kt
package com.ggardet.hospital.internal.entity

import jakarta.persistence.*
import java.util.*

@Entity
@Table(name = "treatment", schema = "hospitals")
data class Treatment(
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    var id: UUID? = null,

    @Column(nullable = false)
    var name: String,

    @Column
    var description: String? = null,

    @Column
    var durationMinutes: Int = 0,

    @Column
    var protocol: String? = null,

    @ManyToOne
    @JoinColumn(name = "hospital_id")
    var hospital: Hospital? = null
)